### PR TITLE
Let cabal-ghc901.project be cabal.project.local

### PR DESCRIPTION
**Pull request by jneira**
* The intent of the change is let users do `cp cabal-ghc901.project cabal.project.local` to make the workflow smoother
  * but the file can be used in its own as before
* To do so the packages should be uncommented (as they are so in main cabal.project) and the packages without support for ghc-9 should be make buildable with `allow-newer` (thanks to @gbaz for the suggestion)

<a href="https://gitpod.io/#https://github.com/haskell/haskell-language-server/pull/2241"></a>

### CODE REVIEW

**Comment by pepeiborra** [comment]
Can we just replace cabal.project with cabal-ghc901.project ?

**Comment by jneira** [comment]
> 
> 
> Can we just replace cabal.project with cabal-ghc901.project ?

🤔 hmm i had not thought in this possibility, but it would be right one if it is possible, will try, thanks!

**Comment by jneira** [comment]
Possible caveats:
- Source repository packages not compatible with older versions of ghc. I think it is not the case
- Allow newers causing build errors
- Allow newer shadowing errors wich will arise when the version will be released in hackage

But it worths to investigate

**Comment by jneira** [comment]
> 
> 
> Can we just replace cabal.project with cabal-ghc901.project ?

But maybe we could merge this as is? merge both cabal.project is not trivial

**Comment by jneira** [comment]
Other alternative is convert cabal-ghc901.project in cabal-ghc901.local and remove duplication

**Comment by jneira** [comment]
> 
> 
> Other alternative is convert cabal-ghc901.project in cabal-ghc901.local and remove duplication

I think it will be the best option, implementing

**Comment by pepeiborra** [comment]
> > 
> 
> > 
> 
> > Other alternative is convert cabal-ghc901.project in cabal-ghc901.local and remove duplication
> 
> 
> 
> I think it will be the best option, implementing

This might break ghcup /cc @maerwald

File: .gitlab-ci.yml [hunk 1]
```diff
@@ -93,7 +89,6 @@ build-aarch64-darwin:
         --keep CI_PROJECT_DIR \
         --keep MACOSX_DEPLOYMENT_TARGET \
         --keep GHC_VERSION \
-        --keep CABAL_PROJECT \
```
Comment (jneira) [inline .gitlab-ci.yml:96]: @hasufell not sure what has to be done here to take in account LOCAL_CABAL_PROJECT

Comment (hasufell) [reply .gitlab-ci.yml:96]: `--keep LOCAL_CABAL_PROJECT`

File: .gitlab/ci.sh [hunk 2]
```diff
@@ -30,6 +30,10 @@ export BOOTSTRAP_HASKELL_ADJUST_CABAL_CONFIG=yes
 
 curl --proto '=https' --tlsv1.2 -sSf https://get-ghcup.haskell.org | sh
 
+if [[ ! -z "$LOCAL_CABAL_PROJECT" ]]; then
```
Comment (hasufell) [inline .gitlab/ci.sh:33]:
```suggestion
if [[ -n "$LOCAL_CABAL_PROJECT" ]]; then
```

File: .gitlab/ci.sh [hunk 3]
```diff
@@ -30,6 +30,10 @@ export BOOTSTRAP_HASKELL_ADJUST_CABAL_CONFIG=yes
...
+    cp $LOCAL_CABAL_PROJECT cabal.project.local
```
Comment (hasufell) [inline .gitlab/ci.sh:34]:
```suggestion
    run cp "$LOCAL_CABAL_PROJECT" cabal.project.local
```

File: .gitlab-ci.yml (hunk 1 above)
Comment (hasufell) [reply .gitlab-ci.yml:96]: Also, this will break the ci.sh script which still uses this variable.

**Review by hasufell — CHANGES_REQUESTED** [review]
`ci.sh` needs to be fixed

**Comment by jneira** [comment]
Testing gitlab here: https://gitlab.haskell.org/jneira/haskell-language-server/-/jobs/796348

**Comment by jneira** [comment]
@hasufell finally i've kept the CABAL_PROJECT in the script, maybe it is neededin the future

**Comment by jneira** [comment]
I ha

> 
> 
> Testing gitlab here: https://gitlab.haskell.org/jneira/haskell-language-server/-/jobs/796348

```
.gitlab/ci.sh: line 33: CABAL_PROJECT: unbound variable
Cleaning up file based variables 00:01
ERROR: Job failed: exit code 
```

ugh

File: .gitlab/ci.sh [hunk 4]
```diff
@@ -30,6 +30,14 @@ export BOOTSTRAP_HASKELL_ADJUST_CABAL_CONFIG=yes
 
 curl --proto '=https' --tlsv1.2 -sSf https://get-ghcup.haskell.org | sh
 
+if [[ -n "$CABAL_PROJECT" ]]; then
+    CABAL_PROJECT="cabal.project"
+fi
```
Comment (jneira) [inline .gitlab/ci.sh:35]:
```suggestion
```

**Comment by jneira** [comment]
> 
> 
> > > 
> > 
> > 
> > > 
> > 
> > 
> > > Other alternative is convert cabal-ghc901.project in cabal-ghc901.local and remove duplication
> > 
> > 
> > I think it will be the best option, implementing
> 
> This might break ghcup /cc @maerwald

do you refer to `ghcup compile hls`?

**Comment by jneira** [comment]
it seems ci is ignoring the copied `cabal.project.local` and it is trying to build the disabled plugins anyways

**Comment by jneira** [comment]
Not sure if the deduplication worths change the status quo: the gitlab script assumes this is the way we give support for new ghc's nd `ghcup compile hls` has `--cabal-project` to use it too (you would have to download manually the cabal-ghcx.project.local)
So i am gonna close it, sorry for the time spend on this

**Pull request closed without merging**